web/src/app/selection/DestinationInputDirect.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useEffect, useState } from 'react'
+import { useQuery, gql } from 'urql'
+import TextField from '@mui/material/TextField'
+import { InputProps } from '@mui/material/Input'
+import { Check, Close } from '@mui/icons-material'
+import InputAdornment from '@mui/material/InputAdornment'
+import makeStyles from '@mui/styles/makeStyles'
+import { DEBOUNCE_DELAY } from '../config'
+import { DestinationFieldConfig, DestinationType } from '../../schema'
+import AppLink from '../util/AppLink'
+
+const isValidValue = gql`
+  query ValidateDestination($input: DestinationFieldValidateInput!) {
+    destinationFieldValidate(input: $input)
+  }
+`
+
+const useStyles = makeStyles({
+  valid: {
+    fill: 'green',
+  },
+  invalid: {
+    fill: 'red',
+  },
+})
+
+const noSuspense = { suspense: false }
+
+function trimPrefix(value: string, prefix: string): string {
+  if (!prefix) return value
+  if (!value) return value
+  if (value.startsWith(prefix)) return value.slice(prefix.length)
+  return value
+}
+
+export type DestinationInputDirectProps = DestinationFieldConfig & {
+  value: string
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
+  destType: DestinationType
+
+  disabled?: boolean
+}
+
+/**
+ * DestinationInputDirect is a text field that allows the user to enter a
+ * destination directly. It supports validation and live feedback.
+ *
+ * You should almost never use this component directly. Instead, use
+ * DestinationField, which will select the correct component based on the
+ * destination type.
+ */
+export default function DestinationInputDirect(
+  props: DestinationInputDirectProps,
+): JSX.Element {
+  const classes = useStyles()
+
+  const [debouncedValue, setDebouncedValue] = useState(props.value)
+
+  // debounce the input
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(props.value)
+    }, DEBOUNCE_DELAY)
+    return () => {
+      clearTimeout(handler)
+    }
+  }, [props.value])
+
+  // check validation of the input phoneNumber through graphql
+  const [{ data }] = useQuery<{ destinationFieldValidate: boolean }>({
+    query: isValidValue,
+    variables: {
+      input: {
+        destType: props.destType,
+        value: debouncedValue,
+        fieldID: props.fieldID,
+      },
+    },
+    requestPolicy: 'cache-first',
+    pause: !props.value || props.disabled || !props.supportsValidation,
+    context: noSuspense,
+  })
+
+  // fetch validation
+  const valid = !!data?.destinationFieldValidate
+
+  let adorn
+  if (!props.value || !props.supportsValidation) {
+    // no adornment if empty
+  } else if (valid) {
+    adorn = <Check className={classes.valid} />
+  } else if (valid === false) {
+    adorn = <Close className={classes.invalid} />
```

Comment:
Let's use the sx prop instead, and import green & red from material to stay consistent

```suggestion
    adorn = <Check sx={{ fill: green[500] }} />
  } else if (valid === false) {
    adorn = <Close sx={{ fill: red[500] }} />
```